### DISCUSSION TITLE
If you send 8 characters in serial, you will not be able to send serial after that.

### DISCUSSION THREAD

**Discussion by INOUETAICHI** [stm32duino/Arduino_Core_STM32#1871] [Q&A] [answered]
[environment]
macOS Monterey 12.5.1（21G83）
MacBook Air (M1, 2020)

[bug]
If you send the following string from the Arduino serial monitor, you will not be able to send serial after that
"12345678"
"1234567\n"
"eightchr"

```
void setup() {
  Serial.begin(115200);
}
void loop() {
  if (Serial.available())
    Serial.write(Serial.read());
}
```

I'm sorry if this isn't something to discuss here.

**Comment by fpistm** [comment]
I've tested and works. Which board you used?

**Comment by INOUETAICHI** [comment]
Nucleo F446RE
https://www.st.com/ja/evaluation-tools/nucleo-f446re.html

**Comment by INOUETAICHI** [comment]
It also occur on MacBook Air M1 2020 / Ventura 13.0.1.

**Comment by fpistm** [comment, marked as answer]
Maybe linked to the STLink Firmware which handle the VCP.
Like for the mass storage upload method, could you try to update the STLink Firmware.

**Reply by INOUETAICHI** [reply]
I upgraded stlink firmware from V2.J29.M18 to V2.J40.M27 and solved!
Thank you so much fpistm!!
